common/utils/Clock.cpp

```diff
@@ -251,12 +263,26 @@ const TimeStamp TimeStamp::operator-(const TimeInterval &interval) const {
   return TimeStamp(m_tv - interval.m_interval);
 }
 
-void Clock::CurrentTime(TimeStamp *timestamp) const {
+void Clock::CurrentMonotonicTime(TimeStamp *timestamp) const {
+#ifdef CLOCK_MONOTONIC
+  struct timespec ts;
+  clock_gettime(CLOCK_MONOTONIC, &ts);
+  *timestamp = ts;
+#else
+  CurrentRealTime(timestamp);
+#endif
+}
+
+void Clock::CurrentRealTime(TimeStamp *timestamp) const {
   struct timeval tv;
   gettimeofday(&tv, NULL);
   *timestamp = tv;
 }
 
+void Clock::CurrentTime(TimeStamp *timestamp) const {
+  CurrentMonotonicTime(timestamp);
```

Comment:
I'm just thinking, this should probably be real time for full backwards compatibility.

Reply:
Done.